WebDriverAgentLib/Utilities/FBXCodeCompatibility.m

```diff
@@ -41,5 +46,29 @@ + (instancetype)fb_applicationWithPID:(pid_t)processID
   return [self applicationWithPID:processID];
 }
 
+- (void)fb_activate
+{
+  dispatch_once(&onceActivate, ^{
+    FBCanUseActivate = [self respondsToSelector:@selector(activate)];
+  });
+  if (!FBCanUseActivate) {
+    [[NSException exceptionWithName:FBApplicationMethodNotSupportedException reason:@"'activate' method is not supported by the current iOS SDK" userInfo:@{}] raise];
+  }
+  [self activate];
+}
+
+- (NSUInteger)fb_state
+{
+  return [[self valueForKey:@"state"] intValue];
```

Comment:
What was the point of removing state from XCUIApplicaiton?

Reply:
the signature of that method has changed since Xcode9, so the compiler was complaining about such inconsistence.

Reply:
How different? They changed `NSUInteger` to `NSInteger` ?